File: unidbg-android/src/main/java/com/vortexdbg/linux/android/dvm/wrapper/DvmBoolean.java
package com.vortexdbg.linux.android.dvm.wrapper;

import com.vortexdbg.linux.android.dvm.DvmObject;
import com.vortexdbg.linux.android.dvm.VM;

public class DvmBoolean extends DvmObject<Boolean> {

    @SuppressWarnings("unused")
    public static DvmBoolean valueOf(VM vm, boolean b) {
        return new DvmBoolean(vm, b);
    }

    private DvmBoolean(VM vm, Boolean value) {
        super(vm.resolveClass("java/lang/Boolean"), value);
    }

    @Override
    public String toString() {
        return Boolean.toString(value);
    }
}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/android/dvm/wrapper/DvmBoolean.kt
package com.vortexdbg.linux.android.dvm.wrapper

import com.vortexdbg.linux.android.dvm.DvmObject
import com.vortexdbg.linux.android.dvm.VM

class DvmBoolean private constructor(vm: VM, value: Boolean) : DvmObject<Boolean>(vm.resolveClass("java/lang/Boolean"), value) {

    override fun toString(): String {
        return java.lang.Boolean.toString(value)
    }

    companion object {
        @JvmStatic
        @Suppress("unused")
        fun valueOf(vm: VM, b: Boolean): DvmBoolean {
            return DvmBoolean(vm, b)
        }
    }
}
